### PR TITLE
Add --list-package command line to show all supported packages #1643

### PR DESCRIPTION
**Pull request by linexb**
Add --list-package command line to show all supported packages

### CODE REVIEW

**Comment by linexb** [comment]
@pombredanne  This pull request is ready. Please review. Thanks. :)

File: src/scancode/cli.py [hunk 1]
```diff
@@ -62,6 +62,8 @@
 from commoncode.fileutils import POSIX_PATH_SEP
 from commoncode.timeutils import time2tstamp
 
+from packagedcode import PACKAGE_TYPES
```
Comment (pombredanne) [inline src/scancode/cli.py]: IMHO we cannot import this globally... Instead make this an option in the packagedcode plugin and not in the main cli.py

File: src/scancode/cli.py [hunk 2]
```diff
@@ -356,6 +379,12 @@ def print_options(ctx, param, value):
     help='Show the list of available ScanCode plugins and exit.',
     help_group=DOC_GROUP, cls=CommandLineOption)
 
+@click.option('--list-packages',
```
Comment (pombredanne) [inline src/scancode/cli.py]: This should go in the packagedcode module and plugin instead

Comment (linexb) [reply src/scancode/cli.py]: @pombredanne  Got it, thanks. I will update according to this idea.

File: src/scancode/cli.py (hunk 1 above)
Comment (linexb) [reply src/scancode/cli.py]: @pombredanne  Thanks, I have moved the the option in the packagedcode plugin instead.

**Comment by linexb** [comment]
@pombredanne  The pull request is ready. The changes include:
1. Fix according to your comments to move the command in packagedcode plugin
2. Refine the recognize function of python package to make sure python scan works.
3. Add parse function of PKG-INFO file.

**Comment by pombredanne** [comment]
@linexb Thank you!
Merging.

**Pull request merged**